### PR TITLE
Constructor colors refactor

### PR DESCRIPTION
**Pull request by Iryna-Bzdyr**

### CODE REVIEW

File: modules/constructor/constructor-basic/constructor-basic.model.js [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
   },
+  color: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Color',
+  },
   image: String,
   basePrice: [CurrencySet],
-  available: Boolean
+  available: Boolean,
+  default:Boolean,
```
Comment (kapoliub) [inline modules/constructor/constructor-basic/constructor-basic.model.js:18]: is this `default` as key or you want set some field with default value?

File: modules/constructor/constructor-bottom/constructor-bottom.model.js [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
     ref: 'Material',
   },
+  color: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Color',
+  },
   image: String,
   basePrice: [CurrencySet],
   available: Boolean,
+  default:Boolean,
```
Comment (kapoliub) [inline modules/constructor/constructor-bottom/constructor-bottom.model.js:18]: same

File: modules/constructor/constructor-front-pocket/constructor-front-pocket.model.js [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
   },
+  color: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Color',
+  },
   image: String,
   basePrice: [CurrencySet],
-  available: Boolean
+  available: Boolean,
+  default:Boolean,
```
Comment (kapoliub) [inline modules/constructor/constructor-front-pocket/constructor-front-pocket.model.js:18]: same

File: modules/model/model.service.js [hunk 4]
```diff
@@ -33,39 +33,36 @@ class ModelsService {
...
-        populate: {
-          path: 'color',
-          model: 'Color',
-        },
-      },
+        model: 'Material'
+      }, {
+        path: 'color',
+        model: 'Color'
+      }]
```
Comment (kapoliub) [inline modules/model/model.service.js:65]: you can use one `populate` method with the array of objects

**Review by dieie32 — APPROVED** [review]

File: modules/constructor/constructor-basic/constructor-basic.model.js (hunk 1 above)
Comment (Iryna-Bzdyr) [reply modules/constructor/constructor-basic/constructor-basic.model.js:18]: Thit id default as a key in order you want use this elemend by defauld when counstructor is loaded for the first time

File: modules/constructor/constructor-bottom/constructor-bottom.model.js (hunk 2 above)
Comment (Iryna-Bzdyr) [reply modules/constructor/constructor-bottom/constructor-bottom.model.js:18]: Thit id default as a key in order you want use this elemend by defauld when counstructor is loaded for the first time

File: modules/constructor/constructor-front-pocket/constructor-front-pocket.model.js (hunk 3 above)
Comment (Iryna-Bzdyr) [reply modules/constructor/constructor-front-pocket/constructor-front-pocket.model.js:18]: Thit is default as a key in order you want use this element by defauld when counstructor is loaded for the first time

File: modules/model/model.service.js (hunk 4 above)
Comment (Iryna-Bzdyr) [reply modules/model/model.service.js:65]: This isn't array of object  it is parametrs of populate method which you can pass as an array. In order to write this
populate(material).populate(color) you can use one populate method and then pass all parametrs as an array of objects whith keys path and model. It's work correctly in both cases

**Review by ivan-bonk — APPROVED** [review]

File: modules/model/model.service.js (hunk 4 above)
Comment (kapoliub) [reply modules/model/model.service.js:65]: you can use populate like this `populate([
  {path: 'material, model: 'Material', 
     populate:{ 
       path: 'somePath', model: 'someModel'
    }
  }, 
  {path: 'color', model: 'Colors'}])`

**Review by kapoliub — APPROVED** [review]

File: modules/model/model.service.js (hunk 4 above)
Comment (Iryna-Bzdyr) [reply modules/model/model.service.js]: Fixed

**Pull request merged**